src/main/java/com/example/demo/service/file/CustomFileReader.java

```diff
@@ -0,0 +1,27 @@
+package com.example.demo.service.file;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomFileReader {
```

Comment:
this class implements reading file from file system

we can create some abstract FileReader interface and make this class implement it (and call it FileSystemFileReader, for example)

and then use the abstract type for injection

Reply:
done